### ISSUE TITLE
Guide to showcase the quickest way to create a package based on a C program

### ISSUE THREAD

**Issue by greenbreakfast** [OnionIoT/documentation#171]
## Motivation
*Why are we doing this? What use cases does it support? What problems does it solve? What is the expected outcome?*

C programs are 
This helps streamline the development process on the Omega2

### **Is your documentation request related to a problem? Please describe.**
*A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]*

Onion recommends wrapping all custom code/applications into packages.
But there isn't a single source that can be followed from start to finish that explains how to create a package based on a C program

## **Describe the solution you'd like**
*A clear and concise description of what you want to happen.*

A example-based, step by step guide that's easy to follow that shows me how to create a package based on a C program. It should mention the environment setup but ideally this should be very easy and quick.

I want to know:
- how to compile the package
- transfer the package to Omega
- Install the apckage
- Run the installed application

## **Describe alternatives you've considered**
*A clear and concise description of any alternative solutions or features you've considered.*

- for defining a package: working backwards from existing packages, and referencing the [create a custom package article](https://documentation.onioniot.com/packages/custom-package)
- for compiling: following the [how to compile a package article](https://documentation.onioniot.com/packages/compile-package) and trying to piece everything together

It would be ideal to have everything in one place, focusing on an example. Once I complete the example, I should be able to adapt it to my own application

**Comment by greenbreakfast** [comment]
Note for implementation: A new "Guides" section to the documentation will be added.
To publish the guide, please work in the `content/guides` branch on the `guides/packages/c-package-example.md` file

**Comment by remingtontr** [comment]
OKay.


Tom Remington

________________________________

**Comment by remingtontr** [comment]
Ready for initial review.

**Comment by remingtontr** [comment]
I have revised the article based on comments and new content in the outline file. Please review when you get the chance.


Tom Remington

________________________________

**Comment by greenbreakfast** [comment]
@remingtontr thanks! added another round of comments and suggestions

**Comment by remingtontr** [comment]
Updated!


Tom Remington

________________________________

**Comment by remingtontr** [comment]
Updated.

Tom Remington

________________________________

**Comment by remingtontr** [comment]
Okay, so I added 171 directly to the contents/guides branch.

**Comment by remingtontr** [comment]
I got ahead of myself and committed changes for this on the content/guides branch directly rather than creating an Issue. My commit is under the  #177 issue. Please let me know if you want me to back this out and create a separate PR.

**Comment by greenbreakfast** [comment]
@remingtontr yep, committing to the content/guides branch is ok. I made PR #177 so we can see it rendered. 
I will take a closer look later this week and get back to you

**Comment by remingtontr** [comment]
Lazar,

Sounds good!


Tom Remington

________________________________

**Comment by greenbreakfast** [comment]
completed in PR #177 - closing this issue
